azurerm/resource_arm_netapp_volume.go

```diff
@@ -0,0 +1,357 @@
+		Pending: []string{"200", "202"},
+		Target:  []string{"404"},
+		Refresh: netappVolumeDeleteStateRefreshFunc(ctx, client, resourceGroup, accountName, poolName, name),
+		Timeout: 20 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for NetApp Volume Provisioning Service %q (Resource Group %q) to be deleted: %+v", name, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func netappVolumeDeleteStateRefreshFunc(ctx context.Context, client *netapp.VolumesClient, resourceGroupName string, accountName string, poolName string, name string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		res, err := client.Get(ctx, resourceGroupName, accountName, poolName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(res.Response) {
+				return nil, "", fmt.Errorf("Error retrieving NetApp Volume %q (Resource Group %q): %s", name, resourceGroupName, err)
+			}
+		}
+
+		if _, err := client.Delete(ctx, resourceGroupName, accountName, poolName, name); err != nil {
+			log.Printf("Error reissuing NetApp Volume %q delete request (Resource Group %q): %+v", name, resourceGroupName, err)
+		}
+
+		return res, strconv.Itoa(res.StatusCode), nil
+	}
+}
+
+func expandArmExportPolicyRule(input []interface{}) *netapp.VolumePropertiesExportPolicy {
+	results := make([]netapp.ExportPolicyRule, 0)
+	for _, item := range input {
+		if item != nil {
+			v := item.(map[string]interface{})
+			ruleIndex := int32(v["rule_index"].(int))
+			allowedClients := v["allowed_clients"].(string)
+			cifs := v["cifs"].(bool)
+			nfsv3 := v["nfsv3"].(bool)
+			nfsv4 := v["nfsv4"].(bool)
+			unixReadOnly := v["unix_read_only"].(bool)
+			unixReadWrite := v["unix_read_write"].(bool)
+
+			result := netapp.ExportPolicyRule{
+				AllowedClients: utils.String(allowedClients),
+				Cifs:           utils.Bool(cifs),
+				Nfsv3:          utils.Bool(nfsv3),
+				Nfsv4:          utils.Bool(nfsv4),
+				RuleIndex:      utils.Int32(ruleIndex),
+				UnixReadOnly:   utils.Bool(unixReadOnly),
+				UnixReadWrite:  utils.Bool(unixReadWrite),
+			}
+
+			results = append(results, result)
+		}
+	}
+
+	result := netapp.VolumePropertiesExportPolicy{
+		Rules: &results,
+	}
+
+	return &result
+}
+
+func flattenArmExportPolicyRule(input *[]netapp.ExportPolicyRule) []interface{} {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results
+	}
+
+	for _, item := range *input {
+		c := make(map[string]interface{})
+
+		if v := item.RuleIndex; v != nil {
+			c["rule_index"] = *v
+		}
+		if v := item.AllowedClients; v != nil {
+			c["allowed_clients"] = *v
+		}
+		if v := item.Cifs; v != nil {
+			c["cifs"] = *v
+		}
+		if v := item.Nfsv3; v != nil {
+			c["nfsv3"] = *v
+		}
+		if v := item.Nfsv4; v != nil {
+			c["nfsv4"] = *v
+		}
+		if v := item.UnixReadOnly; v != nil {
+			c["unix_read_only"] = *v
+		}
+		if v := item.UnixReadWrite; v != nil {
+			c["unix_read_write"] = *v
+		}
```

Comment:
all of these values need to be set in all cases; as such can we update this to be:
]
```suggestion
		unixReadWrite := false
		if v := item.UnixReadWrite; v != nil {
			unixReadWrite = *v
		}
		
		...
		
		results = append(results, map[string]interface{}{
			"unix_read_write": unixReadWrite,
		})
```

Reply:
I don't think so. We can left it empty.
![image](https://user-images.githubusercontent.com/19754191/69304562-08a86680-0c5c-11ea-871d-f4ac764219bc.png)

Reply:
this is a Terraform specific thing where all inner fields need to be specified

Reply:
so you mean I don’t need to change the code here as you suggested, right?

Reply:
If I use your suggested code, it would add at least one item. Then the behavior is different with the resource in azure portal because the behavior of this resource in azure portal allows not to add any item. So we need to keep align with azure portal, then I don’t need to change the code as you suggested, right?

Reply:
fixed